Crash on Forge (via Sinytra Connector) when Connector Extras is installed

The mod crashes when running on Forge under [Sinytra Connector](https://github.com/Sinytra/Connector) when [Connector Extras](https://github.com/Sinytra/ConnectorExtras) is also installed. This is because Connector Extras [provides](https://github.com/Sinytra/ConnectorExtras/blob/5b22b019f3027067e8f0771ba7f7cf542b659296/modmenu-bridge/src/main/resources/META-INF/mods.toml#L18) the `modmenu` mod ID, but [doesn't re-implement](https://github.com/Sinytra/ConnectorExtras/tree/5b22b019f3027067e8f0771ba7f7cf542b659296/modmenu-bridge/src/main/java/com/terraformersmc/modmenu) non-API classes of Mod Menu; only its config API.

<details>

<summary>Stacktrace (with emogg)</summary>

```
net.minecraftforge.fml.loading.EarlyLoadingException: Could not execute entrypoint stage 'client' due to errors, provided by 'emogg'!
	at dev.su5ed.sinytra.connector.loader.ConnectorEarlyLoader.createLoadingException(ConnectorEarlyLoader.java:81) ~[Connector-1.0.0-beta.36+1.20.1.jar%23718!/:1.0.0-beta.36+1.20.1] {}
	at dev.su5ed.sinytra.connector.loader.ConnectorEarlyLoader.createGenericLoadingException(ConnectorEarlyLoader.java:77) ~[Connector-1.0.0-beta.36+1.20.1.jar%23718!/:1.0.0-beta.36+1.20.1] {}
	at dev.su5ed.sinytra.connector.mod.ConnectorLoader.load(ConnectorLoader.java:55) ~[Connector-1.0.0-beta.36+1.20.1-mod.jar%23723!/:1.0.0-beta.36+1.20.1] {re:mixin,re:classloading}
	at net.minecraft.client.Minecraft.handler$eig000$connectormod$earlyInit(dev/su5ed/sinytra/connector/mod/mixin/boot/MinecraftMixin.java [connectormod.mixins.json]:15) ~[client-1.20.1-20230612.114412-srg.jar%231294!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraft_runtick,xf:fml:xaeroworldmap:xaero_wm_minecraft_runtick_render_call,pl:connector_pre_launch:A,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraft_runtick,xf:fml:xaeroworldmap:xaero_wm_minecraft_runtick_render_call,pl:mixin:A,pl:connector_pre_launch:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.<init>(Minecraft.java:424) ~[client-1.20.1-20230612.114412-srg.jar%231294!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraft_runtick,xf:fml:xaeroworldmap:xaero_wm_minecraft_runtick_render_call,pl:connector_pre_launch:A,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraft_runtick,xf:fml:xaeroworldmap:xaero_wm_minecraft_runtick_render_call,pl:mixin:A,pl:connector_pre_launch:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:182) ~[minecraft-1.20.1-client.jar:?] {re:mixin,pl:connector_pre_launch:A,pl:runtimedistcleaner:A,re:classloading,pl:mixin:APP:flywheel.mixins.json:ClientMainMixin from mod flywheel,pl:mixin:APP:mixins.emibridge.json:MainEntrypointMixin from mod connectorextras_emi_bridge,pl:mixin:A,pl:connector_pre_launch:A,pl:runtimedistcleaner:A}
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Method.java:580) ~[?:?] {}
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:126) ~[loader-47.1.47.jar:47.1] {}
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:114) ~[loader-47.1.47.jar:47.1] {}
	at net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.runService(CommonClientLaunchHandler.java:24) ~[loader-47.1.47.jar:47.1] {}
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:108) ~[loader-47.1.47.jar:47.1] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?] {}
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Method.java:580) ~[?:?] {}
	at io.github.zekerzhayard.forgewrapper.installer.Main.main(Main.java:67) ~[?:?] {}
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87) ~[?:?] {}
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:125) ~[?:?] {}
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) ~[?:?] {}
Caused by: java.lang.RuntimeException: Could not execute entrypoint stage 'client' due to errors, provided by 'emogg'!
	at net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$0(FabricLoaderImpl.java:131) ~[Connector-1.0.0-beta.36+1.20.1.jar%23718!/:1.0.0-beta.36+1.20.1] {}
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33) ~[Connector-1.0.0-beta.36+1.20.1.jar%23718!/:1.0.0-beta.36+1.20.1] {}
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:129) ~[Connector-1.0.0-beta.36+1.20.1.jar%23718!/:1.0.0-beta.36+1.20.1] {}
	at dev.su5ed.sinytra.connector.mod.ConnectorLoader.load(ConnectorLoader.java:46) ~[Connector-1.0.0-beta.36+1.20.1-mod.jar%23723!/:1.0.0-beta.36+1.20.1] {re:mixin,re:classloading}
	... 23 more
Caused by: java.lang.NoClassDefFoundError: com/terraformersmc/modmenu/ModMenu
	at io.github.aratakileo.elegantia.util.ModInfo.setConfigScreenGetter(ModInfo.java:78) ~[elegantia-0.0.1-alpha+1.20-1.20.1_mapped_srg_1.20.1.jar%232915!/:?] {re:classloading}
	at io.github.aratakileo.elegantia.gui.config.Config.init(Config.java:324) ~[elegantia-0.0.1-alpha+1.20-1.20.1_mapped_srg_1.20.1.jar%232915!/:?] {re:classloading}
	at io.github.aratakileo.emogg.Emogg.onInitializeClient(Emogg.java:70) ~[emogg-2.0.0+1.20-1.20.1-forged_mapped_srg_1.20.1.jar%232916!/:?] {re:mixin,re:classloading}
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:127) ~[Connector-1.0.0-beta.36+1.20.1.jar%23718!/:1.0.0-beta.36+1.20.1] {}
	at dev.su5ed.sinytra.connector.mod.ConnectorLoader.load(ConnectorLoader.java:46) ~[Connector-1.0.0-beta.36+1.20.1-mod.jar%23723!/:1.0.0-beta.36+1.20.1] {re:mixin,re:classloading}
	... 23 more
Caused by: java.lang.ClassNotFoundException: com.terraformersmc.modmenu.ModMenu
	at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[?:?] {}
	at java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?] {}
	at cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:137) ~[securejarhandler-2.1.10.jar:?] {}
	at java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?] {}
	at cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:137) ~[securejarhandler-2.1.10.jar:?] {}
	at java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?] {}
	at io.github.aratakileo.elegantia.util.ModInfo.setConfigScreenGetter(ModInfo.java:78) ~[elegantia-0.0.1-alpha+1.20-1.20.1_mapped_srg_1.20.1.jar%232915!/:?] {re:classloading}
	at io.github.aratakileo.elegantia.gui.config.Config.init(Config.java:324) ~[elegantia-0.0.1-alpha+1.20-1.20.1_mapped_srg_1.20.1.jar%232915!/:?] {re:classloading}
	at io.github.aratakileo.emogg.Emogg.onInitializeClient(Emogg.java:70) ~[emogg-2.0.0+1.20-1.20.1-forged_mapped_srg_1.20.1.jar%232916!/:?] {re:mixin,re:classloading}
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:127) ~[Connector-1.0.0-beta.36+1.20.1.jar%23718!/:1.0.0-beta.36+1.20.1] {}
	at dev.su5ed.sinytra.connector.mod.ConnectorLoader.load(ConnectorLoader.java:46) ~[Connector-1.0.0-beta.36+1.20.1-mod.jar%23723!/:1.0.0-beta.36+1.20.1] {re:mixin,re:classloading}
	... 23 more
```

</details>

The issue is here:
https://github.com/aratakileo/elegantia/blob/337ecca7be91bb467df634b7e350d7ddde3d1e78/src/main/java/io/github/aratakileo/elegantia/util/ModInfo.java#L74-L79

It appears that reflecting a non-API class of Mod Menu like this is not entirely safe. Could there be an alternative? ( or at least a check for `if (!isModLoaded("connectormod")) return;` :p )

--- a/src/main/java/io/github/aratakileo/elegantia/util/ModInfo.java
+++ b/src/main/java/io/github/aratakileo/elegantia/util/ModInfo.java
@@ -45,6 +45,19 @@ public abstract class ModInfo {
     public void setConfigScreenGetter(@NotNull Function<Screen, Screen> configScreenGetter) {
         if (!isModLoaded("modmenu")) return;
 
+        /*
+         * This check is necessary to avoid crashing when launching the mod by Forge/Neoforge in situations:
+         *  - Sinytra Connector is installed without Forgified Fabric API
+         *  - Sinytra Connector and Connector Extras are installed
+         *
+         * NOTE: Mod with id `fabric_api` is Forgified Fabric API
+         */
+        if (
+                isModLoaded("connectormod")
+                        && !isModLoaded("fabric_api")
+                        || isModLoaded("connectorextras")
+        ) return;
+
         try {
             /*
              * Adding configuration screens is necessary in this way, due to the fact

